src/Microsoft.OData.Core/ODataBatchReader.cs

```diff
@@ -108,12 +101,62 @@ public ODataBatchReaderState State
                 return this.batchReaderState;
             }
 
-            private set
+            set

```

Comment:
>set [](start = 12, length = 3)

set should be private (ideally) or (minimally) protected.  The consumer of the batch reader should never be able to set the state.

Reply:
OK. It can be made protected.
It cannot be private. It is not designed to be encapsulated by the base class here since Reader state needs to be set by derived class such as MIME batch reader depending on batch-type-specific handling.